### PR TITLE
Remove repr_converter

### PR DESCRIPTION
**Pull request by graingert**
There's some remnants of py2 compat shims still in the codebase, one of which is repr_converter

In this PR I  remove those remnants of py2/py3 compat code.

However rather than replace repr_converter with to_unicode, I removed it.

Currently the repr_converter causes pureber etc to lie about
their content, worse when provided bytes that cannot be
decoded with utf8 (eg ObjectSID)
the reprs crash causing problems with pytest assertions

### Contributor Checklist:

- [ ] I have updated the release notes at `docs/source/NEWS.rst`
- [x] I have updated the automated tests.
- [ ] All tests pass on your local dev environment. See `CONTRIBUTING.rst`.

### CODE REVIEW

File: ldaptor/ldiftree.py [hunk 1]
```diff
@@ -289,14 +289,12 @@ def __repr__(self):
 
     def __lt__(self, other):
         if not isinstance(other, LDIFTreeEntry):
-            # We don't return NotImplemented so that we get the same
-            # result in Python2 and Python3.
-            raise (TypeError, "unorderable types: {!r} > {!r}".format(self, other))
+            return NotImplemented
```
Comment (adiroiban) [inline ldaptor/ldiftree.py:292]: Maybe have a better error message here
```suggestion
            return NotImplemented('LDIFTreeEntry comparison operation not supported.')
```

File: ldaptor/ldiftree.py [hunk 2]
```diff
@@ -289,14 +289,12 @@ def __repr__(self):
...
         return self.dn < other.dn
 
     def __gt__(self, other):
         if not isinstance(other, LDIFTreeEntry):
-            raise (TypeError, "unorderable types: {!r} < {!r}".format(self, other))
+            return NotImplemented
```
Comment (adiroiban) [inline ldaptor/ldiftree.py:297]: same as above regarding adding a more informative error message.

File: ldaptor/protocols/pureber.py [hunk 3]
```diff
@@ -225,7 +225,7 @@ def toWire(self):
         return result
 
     def __repr__(self):
-        value = repr_converter(self.value)
+        value = self.value
```
Comment (adiroiban) [inline ldaptor/protocols/pureber.py:228]: this is ok... but maybe have a single representation for all instances

I have no idea why we need a special case here.

```
return self.__class__.__name__ + "(value=%r, tag=%d)" % (value, self.tag)
```

**Review by adiroiban — APPROVED** [review]
Looks good.

I am a bit worried that we have some many repr tests and we have the special handling depending on the tag value.

repr should only be used for debugging and as dev helper tools.. and we should only have the minium test coverage for repr ... and the returned value should not be part of the public API

File: ldaptor/ldiftree.py (hunk 1 above)
Comment (graingert) [reply ldaptor/ldiftree.py:292]: `TypeError: 'NotImplementedType' object is not callable`

**Comment by adiroiban** [comment]
Regarding required coverage.
I made the following coverage checks as required:

* codecov/patch - a diff should have full coverage as any new code should be merged with associated tests
* codecov/project/tests - make sure all tests are executed at least once. this should detect if we have tests which are always skipped

File: ldaptor/protocols/pureber.py (hunk 3 above)
Comment (graingert) [reply ldaptor/protocols/pureber.py:228]: probably going to attrs everything next week, dunno why tag is sometimes excluded.

File: ldaptor/ldiftree.py (hunk 1 above)
Comment (adiroiban) [reply ldaptor/ldiftree.py:292]: Sorry. Will  `NotImplementedError` do the job?

Comment (graingert) [reply ldaptor/ldiftree.py:292]: NotImplemented automatically raises a              `raise (TypeError, "unorderable types: {!r} > {!r}".format(self, other)) ` style message with nice reprs etc - it's just a bit faster

Comment (graingert) [reply ldaptor/ldiftree.py:292]: >  If all attempts return NotImplemented, the interpreter will raise an appropriate exception.

**Pull request merged**